orttraining/orttraining/training_ops/cuda/nn/dropout.cc

```diff
@@ -9,6 +9,17 @@
 namespace onnxruntime {
 namespace cuda {
 
+#if defined(CUDA_VERSION) && CUDA_VERSION >= 11000
+#define ALL_IEEE_FLOAT_TENSOR_TYPES {DataTypeImpl::GetTensorType<float>(),      \
+                                     DataTypeImpl::GetTensorType<double>(),     \
+                                     DataTypeImpl::GetTensorType<MLFloat16>(),  \
+                                     DataTypeImpl::GetTensorType<BFloat16>()}
+#define ALL_IEEE_FLOAT_DATA_TYPES float, MLFloat16, double, BFloat16
+#else
+#define ALL_IEEE_FLOAT_TENSOR_TYPES DataTypeImpl::AllIEEEFloatTensorTypes()
+#define ALL_IEEE_FLOAT_DATA_TYPES float, MLFloat16, double
+#endif
+
```

Comment:
Should we refactor https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/core/framework/data_types.cc#L898-L905 to prevent code duplicate or implementation discrepancies?

Reply:
This is just merged from master, not related to this PR.